cmake/Components.cmake

```diff
@@ -0,0 +1,146 @@
+include(CMakeParseArguments)
+
+set(global_prefix "${PROJECT_NAME}_")
+set(component_prefix "${PROJECT_NAME}_component_")
+
+define_property(GLOBAL PROPERTY "${global_prefix}INCLUDE_DIRS"
+  BRIEF_DOCS "Global include directories used by all components."
+  FULL_DOCS "Global include directories used by all components."
+)
+define_property(GLOBAL PROPERTY "${global_prefix}COMPONENTS"
+  BRIEF_DOCS "List all known Aikido components."
+  FULL_DOCS "List all known Aikido components."
+)
+
+#==============================================================================
+function(add_component component)
```

Comment:
Well, I don't think we should necessarily assume that we are making components for something named `PROJECT_NAME`.

I think one reasonable way to do this would be to actually add another property to these functions like:
```cmake
function(add_component package_name component)
    [...]
endfunction()

macro(setup_component_helpers package_name)
    function(add_${package_name}_component)
        add_component(PACKAGE aikido ${ARGV}})
    endfunction()
    [...]
endmacro()
```
and then in _aikido_'s top-level cmake do something like:
```cmake
include(Components.cmake)
setup_component_helpers(aikido)

add_aikido_component(...)
```
(based on http://stackoverflow.com/a/31947751)